### PR TITLE
bug(plunker): fix plunkers not working with dialogs #5899

### PR DESCRIPTION
**Pull request by amcdnl**
Fixes: https://github.com/angular/material2/issues/5899

### CODE REVIEW

**Comment by julianobrasil** [comment]
@amcdnl, I haven't seen any mention in the code to the `entryComponents` array. It should be present in the case of `MdDialog`.

**Comment by jelbourn** [comment]
@tinayuangao is better to review since she authored the plunker generation

File: src/app/shared/plunker/plunker-writer.ts [hunk 1]
```diff
@@ -113,9 +113,25 @@ export class PlunkerWriter {
...
+
+      // Replace `import {MaterialDocsExample} from 'material-docs-example'`
       // will be replaced as `import {ButtonDemo} from './button-demo'`
-      fileContent = fileContent.replace(/MaterialDocsExample/g, data.componentName);
+      fileContent = fileContent.replace(/{MaterialDocsExample}/g, `{${data.componentName}}`);
+
+      // Replace `declarations: [MaterialDocsExample]`
+      // will be replaced as `declarations: [ButtonDemo]`
+      fileContent = fileContent.
+        replace(/declarations: \[MaterialDocsExample\]/g,
```
Comment (tinayuangao) [inline src/app/shared/plunker/plunker-writer.ts]: Also replace `entryComponents`? 

Add `entryComponents: [MaterialDocsExample]` to `main.ts`.

I tested on plunker and found out it doesn't work without entryComponents.

**Comment by amcdnl** [comment]
@tinayuangao - Is this PR still relevant. Looks like they are working today.

**Comment by julianobrasil** [comment]
@tinayuangao, dialogs are still throwing exceptions (lack of entryComponents).

**Comment by amcdnl** [comment]
@julianobrasil where is it throwing the errors?

**Comment by julianobrasil** [comment]
@amcdnl, if you click on the link to the plunk example for the dialogs (at https://material.angular.io/components/dialog/overview), you'll see the errors:

image

Notice that the **`entryComponents`** attribute is missing and there are two components in **`bootstrap`** attribute.

**Comment by emanuelet** [comment]
still same error

**Comment by amcdnl** [comment]
Closing this since it outdated, will update in a new PR.

**Comment by amcdnl** [comment]
Resolved here - https://github.com/angular/material.angular.io/pull/356

**Pull request closed without merging**